public/app/plugins/panel/canvas/InlineEdit.tsx

```diff
@@ -0,0 +1,110 @@
+import { css, cx } from '@emotion/css';
+import React, { SyntheticEvent, useRef, useState } from 'react';
+import Draggable from 'react-draggable';
+import { Resizable, ResizeCallbackData } from 'react-resizable';
+
+import { Dimensions2D, GrafanaTheme } from '@grafana/data';
+import { IconButton, stylesFactory, useTheme } from '@grafana/ui';
+import store from 'app/core/store';
+
+import { InlineEditOptions } from './InlineEditOptions';
+
+type Props = {
+  onClose?: () => void;
+};
+
+const OFFSET = 8;
+
+export const InlineEdit = ({ onClose }: Props) => {
+  const theme = useTheme();
+  const btnInlineEdit = document.querySelector('[data-btninlineedit]')!.getBoundingClientRect();
+  const ref = useRef<HTMLDivElement>(null);
+  const styles = getStyles(theme);
+  const inlineEditKey = 'inlineEditPanel';
+
+  const defaultMeasurements = { width: 350, height: 400 };
+  const defaultX = btnInlineEdit.x - btnInlineEdit.width + OFFSET;
+  const defaultY = -OFFSET - defaultMeasurements.height;
+
+  const savedPlacement = store.getObject(inlineEditKey, {
+    x: defaultX,
+    y: defaultY,
+    w: defaultMeasurements.width,
+    h: defaultMeasurements.height,
+  });
+  const [measurements, setMeasurements] = useState<Dimensions2D>({ width: savedPlacement.w, height: savedPlacement.h });
+  const [placement, setPlacement] = useState({ x: savedPlacement.x, y: savedPlacement.y });
+
+  const onDragStop = (event: any, dragElement: any) => {
```

Comment:
We should be able to figure out the type of these events and type this func accordingly (and avoid `any`) :)

Looks like `event` type is `DraggableEvent` and `dragElement` type is `DraggableData`

Reply:
I know.. :( unfortunately couldn't figure those out. `DraggableEvent `  and `DraggableData` don't seem to work either. I'll keep looking.

Reply:
Hm locally 
```suggestion
  const onDragStop = (event: DraggableEvent, dragElement: DraggableData) => {
```
seems to work for me (if we include those as imports) (line 3)
`import Draggable, { DraggableData, DraggableEvent } from 'react-draggable';`

Reply:
Hm, works - it wasn't recognizing the imports.